public/components/datasources/components/manage/associated_objects/utils/associated_objects_tab_empty.tsx

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { EuiButton, EuiEmptyPrompt, EuiLink, EuiText } from '@elastic/eui';
+import React from 'react';
+import { LoadCacheType } from '../../../../../../../common/types/data_connections';
+
+interface AssociatedObjectsTabEmptyProps {
+  cacheType: LoadCacheType;
+}
+
+export const AssociatedObjectsTabEmpty: React.FC<AssociatedObjectsTabEmptyProps> = (props) => {
+  const { cacheType } = props;
+  const QueryWorkbenchButton = (
+    <EuiButton iconSide="right" onClick={() => {}} iconType="popout">
+      Query Workbench
+    </EuiButton>
+  );
+
+  let titleText;
+  let bodyText;
+  switch (cacheType) {
+    case 'databases':
+      titleText = 'You have no databases in your data source';
+      bodyText = 'Add databases and tables to your data source or use Query Workbench';
+      break;
+    case 'tables':
+      titleText = 'You have no associated objects';
+      bodyText = 'Add tables to your data source or use Query Workbench';
+      break;
+    default:
+      titleText = '';
+      bodyText = '';
+      break;
+  }
+
+  return (
+    <EuiEmptyPrompt
+      body={
+        <>
+          <EuiText>
+            <h4>{titleText}</h4>
+            <p>{bodyText}</p>
+          </EuiText>
+          <EuiLink onClick={() => console.log()} external>
+            Learn more
+          </EuiLink>
```

Comment:
Is this finalized in UI, what's the call to action on empty table? Redirect to workbench?

Reply:
+1 not sure what are the pending items for this PR but when should we turn this PR to ready for review? I'm not sure if code like this are still pending for work.